cmd/clusterctl/api/v1alpha3/metadata_type.go

```diff
@@ -62,12 +69,19 @@ func (m *Metadata) GetReleaseSeriesForVersion(version *version.Version) *Release
 }
 
 // GetReleaseSeriesForContract returns the release series for a given API Version, e.g. `v1alpha4`.
+// If more than one release series use the same contract then the latest newer release series is
+// returned.
 func (m *Metadata) GetReleaseSeriesForContract(contract string) *ReleaseSeries {
+	var rs ReleaseSeries
 	for _, releaseSeries := range m.ReleaseSeries {
 		if contract == releaseSeries.Contract {
-			return &releaseSeries
+			if releaseSeries.newer(rs) {
+				rs = releaseSeries
+			}
 		}
 	}
-
-	return nil
+	if rs.Contract == "" {
```

Comment:
can this ever happen? The only case this could happen is if the passed in contract string is also `""` so that case that check should move up to the top so we don't wast time iterating through all the release series when contract is empty

Reply:
Oh nvm, this is to cover the "contract not found" case. To make this logic easier to read, could you either 1) add a small comment that says `// return nil if no release series matching the contract was found`, or 2) add a `found` variable that is set to true when `contract == releaseSeries.Contract` and change this to `if !found` ?

Reply:
> can this ever happen?
yes, it can. If there are not release series that match the given contract then `rs` is never set to anything and `rs.Contract` will remain the zero value (empty string). 

I realize it is a little uncommon to check like this but this essentially to check if we did or did not find a release series with a matching contract. I do it this way because the return type and `m.ReleaseSeries` types do not match (pointer vs struct).

Reply:
> Oh nvm, this is to cover the "contract not found" case.

yes :) 

Updated the PR. Went with option 2.